release-notes/10.0/preview/preview7/aspnetcore.md

```diff
@@ -2,14 +2,252 @@
+- [Avoid cookie login redirects for known API endpoints](#avoid-cookie-login-redirects-for-known-api-endpoints)
+- [Passkey authentication improvements](#passkey-authentication-improvements)
+- [Support for the .localhost top-level domain](#support-for-the-localhost-top-level-domain)
+- [JSON+PipeReader deserialization support](#jsonpipereader-deserialization-support)
+- [Enhanced validation for classes and records](#enhanced-validation-for-classes-and-records)
+- [Blazor component improvements](#blazor-component-improvements)
+- [Identity metrics](#identity-metrics)
+- [OpenAPI improvements](#openapi-improvements)
 
 ASP.NET Core updates in .NET 10:
 
 - [What's new in ASP.NET Core in .NET 10](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-10.0) documentation.
 - [Breaking changes](https://docs.microsoft.com/dotnet/core/compatibility/10.0#aspnet-core)
 - [Roadmap](https://github.com/dotnet/aspnetcore/issues/59443)
 
-## Feature
+## Configure suppressing exception handler diagnostics
 
-Something about the feature
+A new configuration option has been added to the [ASP.NET Core exception handler middleware](https://learn.microsoft.com/aspnet/core/fundamentals/error-handling#exception-handler-page) to control diagnostic output: `ExceptionHandlerOptions.SuppressDiagnosticsCallback`. This callback is passed context about the request and exception, allowing you to add logic that determines whether the middleware should write exception logs and other telemetry.
+
+This setting is useful when you know an exception is transient, or has been handled by the exception handler middleware, and don't want the error logs written to your observability platform.
+
+Additionally, the middleware's default behavior has changed: it no longer writes exception diagnostics for exceptions handled by `IExceptionHandler`. Based on user feedback, logging handled exceptions at the error level was often undesirable when `IExceptionHandler.TryHandleAsync` returned `true`.
+
+You can revert to the previous behavior by configuring `SuppressDiagnosticsCallback`:
+
+```csharp
+app.UseExceptionHandler(new ExceptionHandlerOptions
+{
+    SuppressDiagnosticsCallback = context => false;
+});
+```
+
+For more information about this breaking change, see https://github.com/aspnet/Announcements/issues/524.
+
+## Avoid cookie login redirects for known API endpoints
+
+By default, unauthenticated and unauthorized requests made to known API endpoints protected by cookie authentication now result in 401 and 403 responses rather than redirecting to a login or access denied URI.
+
+This change was [highly requested](https://github.com/dotnet/aspnetcore/issues/9039), because redirecting unauthenticated requests to a login page doesn't usually make sense for API endpoints which typically rely on 401 and 403 status codes rather than HTML redirects to communicate auth failures.
```

Comment:
```suggestion
By default, unauthenticated and unauthorized requests made to known API endpoints protected by cookie authentication now result in 401 and 403 responses rather than redirecting to a login or access denied URI. Redirecting unauthenticated requests to a login page doesn't usually make sense for API endpoints which typically rely on 401 and 403 status codes rather than HTML redirects to communicate authorization failures.
```